apps/api/src/courses/courses.service.ts

```diff
@@ -399,6 +404,98 @@ export class CoursesService {
     };
   }
 
+  async getBetaCourseById(id: string) {
+    const [course] = await this.db
+      .select({
+        id: courses.id,
+        title: courses.title,
+        imageUrl: sql<string>`${courses.imageUrl}`,
+        category: categories.title,
+        categoryId: categories.id,
+        description: sql<string>`${courses.description}`,
+        courseLessonCount: sql<number>`
+          (SELECT COUNT(*)
+          FROM ${courseLessons}
+          JOIN ${lessons} ON ${courseLessons.lessonId} = ${lessons.id}
+          WHERE ${courseLessons.courseId} = ${courses.id} AND ${lessons.archived} = false)::INTEGER`,
+        state: courses.state,
+        priceInCents: courses.priceInCents,
+        currency: courses.currency,
+        archived: courses.archived,
+      })
+      .from(courses)
+      .innerJoin(categories, eq(courses.categoryId, categories.id))
+      .where(and(eq(courses.id, id)));
+
+    if (!course) throw new NotFoundException("Course not found");
+
+    const courseLessonList = await this.db
+      .select({
+        id: lessons.id,
+        title: lessons.title,
+        description: sql<string>`COALESCE(${lessons.description}, '')`,
+        imageUrl: sql<string>`COALESCE(${lessons.imageUrl}, '')`,
```

Comment:
I added this because the existing code requires this data in other views, this will be replaced when we do the whole course adding management view and then we can remove unnecessary code and clean up things like this